lib/streamlit/ForwardMsgCache.py

```diff
@@ -162,7 +200,43 @@ def has_message_reference(self, msg, session):
         populate_hash_if_needed(msg)
         with self._lock:
```

Comment:
I'm actually not sure we need this lock. The message cache is only modified in the server thread, no?

Reply:
@tvst huh - yeah, you're right. (It's also accessed from a /messages GET endpoint, but I think all routes run on the same thread? I'll double-check and change this assuming that's the case.)

Reply:
(Confirmed - I'll remove the lock!)